src/cargo/util/rustc.rs

```diff
@@ -42,11 +44,28 @@ impl Rustc {
             triple.to_string()
         };
 
+        let mut cmd = util::process(&path);
```

Comment:
Gah of course.

Maybe the call to `probe_target_info` could handle errors differently? That is, we could probe with a "check" target, and if that fails, probe as usual (only propagating the latter error)